### PR TITLE
[UTOPIA-1010] FE - new pdf preview route

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- New PDF preview route. The implementation will become neater. Creating PR to not halt Anthony's progress to styling for UTOPIA-1020.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by binggaofw** [comment]
I guess Anthony will work on styles for this ticket. currently it is a little mess. this ticket will only focus on build the framework for download pdf, so i will approve it

Screenshot 2023-04-12 at 4 59 48 PM

**Review by binggaofw — DISMISSED** [review]

**Comment by kushal-arora-fw** [comment]
> I guess Anthony will work on styles for this ticket. currently it is a little mess. this ticket will only focus on build the framework for download pdf, so i will approve it

Your assumption is correct!

File: src/frontend/src/pages/PiaFormPrintPreview/SignatureSection.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import SignatureBlock from './signatureBlock';
+import { ISignatureBlock } from './interface';
+
+const SignatureSection = (data: ISignatureBlock | null) => {
+  const signatureType: string[] = ['name', 'signature', 'date signed'];
+  function printdata() {
+    console.log(data);
+    console.log('not nested');
+    console.log(data);
+  }
```
Comment (binggaofw) [inline src/frontend/src/pages/PiaFormPrintPreview/SignatureSection.tsx:10]: I guess we can delete printdata function

File: src/frontend/src/pages/PiaFormPrintPreview/SignaturesFull.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import MessageSignatures from './MessageSignatures';
+import SignatureSection from './SignatureSection';
+import { ISignatureBlock } from './interface';
+
+const SignaturesFullPIA = () => {
+  function getSignatureSection(key: string): ISignatureBlock | null {
+    if (key in MessageSignatures.Signatures) {
+      console.log('at parent');
```
Comment (binggaofw) [inline src/frontend/src/pages/PiaFormPrintPreview/SignaturesFull.tsx:8]: can remove these console.log

File: src/frontend/src/pages/PiaFormPrintPreview/SignaturesFull.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      console.log(Object(MessageSignatures.Signatures)[key]);
```
Comment (binggaofw) [inline src/frontend/src/pages/PiaFormPrintPreview/SignaturesFull.tsx:9]: same here

**Review by binggaofw — APPROVED** [review]

**Review by binggaofw — DISMISSED** [review]

**Review by binggaofw — APPROVED** [review]

**Comment by kushal-arora-fw** [comment]
is it just me or the tab values don't show up at all?
@binggaofw  - is pdf and its values showing up for you? Did you run in your local

**Comment by binggaofw** [comment]
> is it just me or the tab values don't show up at all? @binggaofw - is pdf and its values showing up for you? Did you run in your local

just test it, the same, yesterday I paired with Anthony and see it works

**Review by Bottle7 — APPROVED** [review]
🎸 🤘🏼

**Comment by Bottle7** [comment]
I'm not getting the data from the intake populated in the print preview. First screen shows the data, the second shows the preview.

image


image

**Comment by binggaofw** [comment]
now it works in my local

**Review by binggaofw — DISMISSED** [review]

File: src/frontend/src/sass/_printPreview.scss [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+ * The page is sent to print in portrait mode.
+ * For the signature section, the signatures will be printed after the form 
+ * on a new page.
+*/
+
+@media print {
+  @page {
+    size: auto; /* auto is the initial value */
+    margin: 12px auto;
+    padding: 50px;
```
Comment (kushal-arora-fw) [inline src/frontend/src/sass/_printPreview.scss:14]: ahhh.. I accidentally committed this as well. Can you please remove this?

**Review by binggaofw — APPROVED** [review]

**Pull request merged**